nextcord/application_command.py

```diff
@@ -0,0 +1,1361 @@
+            else:
+                raise InvalidCommandType(f"{self.type} is not a handled Application Command type.")
+
+    async def call_invoke_message(self, interaction: Interaction) -> None:
+        """|coro|
+        Calls the callback as a message command using the given interaction.
+
+        Parameters
+        ----------
+        interaction: :class:`Interaction`
+            Interaction to get message data from and pass to the callback.
+        """
+        # TODO: Look into function arguments being autoconverted and given? Arg typed "Channel" gets filled with the
+        #  channel?
+        # Is this kinda dumb? Yeah, but at this time it can only return one message.
+        message = self._handle_resolved_message(list(interaction.data["resolved"]["messages"].values())[0])
+        await self.invoke_message(interaction, message)
+
+    async def invoke_message(self, interaction: Interaction, message: Message, **kwargs: Dict[Any, Any]) -> None:
+        if self._self_argument:
+            await self.callback(self._self_argument, interaction, message, **kwargs)
+        else:
+            await self.callback(interaction, message, **kwargs)
+
+    async def call_invoke_user(self, interaction: Interaction) -> None:
+        """|coro|
+        Calls the callback as a user command using the given interaction.
+
+        Parameters
+        ----------
+        interaction: :class:`Interaction`
+            Interaction to get user or member data from and pass to the callback.
+        """
+        # TODO: Look into function arguments being autoconverted and given? Arg typed "Channel" gets filled with the
+        #  channel?
+        # Is this kinda dumb? Yeah, but at this time it can only return one user.
+        user = self._handle_resolved_user(list(interaction.data["resolved"]["users"].values())[0])
+        if interaction.guild and (member := interaction.guild.get_member(user.id)):
+            await self.invoke_user(interaction, member)
+        else:
+            await self.invoke_user(interaction, user)
+
+    async def invoke_user(self, interaction: Interaction, member: Union[Member, User], **kwargs: Dict[Any, Any]) -> None:
+        """|coro|
+        Invokes the callback with the given interaction, member/user, and any additional kwargs added.
+
+        Parameters
+        ----------
+        interaction: :class:`Interaction`
+            Interaction object to pass to the callback.
+        member: Union[:class:`Member`, :class:`User`]
+            Member/user object to pass to the callback.
+        kwargs
+            Any additional keyword arguments to pass to the callback.
+        """
+        if self._self_argument:
+            await self.callback(self._self_argument, interaction, member, **kwargs)
+        else:
+            await self.callback(interaction, member, **kwargs)
+
+    # Decorators.
+
+    def subcommand(self, name: str = MISSING, description: str = MISSING) -> Callable:
```

Comment:
maybe rename this to sub_command for consistency with the other decorators?

Reply:
A small, yet large change. I want others to okay this before going ahead with it.